src/web/app.py

```
@@ -57,153 +59,450 @@
+                yield event
+
+        # Process events with per-iteration timeout check
+        event_iterator = run_with_timeout()
+        start_time = asyncio.get_event_loop().time()
+
+        async for event in event_iterator:
+            # Check if we've exceeded total timeout
+            elapsed = asyncio.get_event_loop().time() - start_time
+            if elapsed > timeout_seconds:
+                logger.warning(f"ADK execution timeout after {elapsed:.1f}s")
+                raise asyncio.TimeoutError(f"Search timed out after {timeout_seconds}s")
```

Ruff (0.14.6)

Abstract `raise` to an inner function

(TRY301)